### DISCUSSION TITLE
Use persist middleware with Slice pattern

### DISCUSSION THREAD

**Discussion by wowblvck** [pmndrs/zustand#2441] [Q&A]
I'm trying to connect persist to my repository using this design:

```tsx
const useStoreBase = create<CombinedState>()(
  persist(
    immer((...api) => ({
      a: createSliceA(...api),
      b: createSliceB(...api),
      c: createSliceC(...api),
    })),
    {
      name: 'store',
      storage: createJSONStorage(() => localStorage),
    },
  ),
);

export const useStore = createSelectors(useStoreBase);
```

And it works fine until I reload the page. After reloading the page, the methods for each of the Slices disappear and an error occurs:
```console
Uncaught TypeError: [some function] is not a function
```
An error occurs when calling a method on any slice from the store.

For example:
```tsx
const sliceA = useStore((state) => state.a);

sliceA.call() - Uncaught TypeError: sliceA.call() is not a function
```

I tried to find a solution here, but alas, I could not find similar cases.

**Comment by dai-shi** [comment]
> After reloading the page, the methods for each of the Slices disappear

This happens because you don't merge your objects (slices). Doesn't it fail without `persist`? Or, maybe `persist` does something. In either case, you are responsible to merge state.

You might want to create a minimal reproduction without `persist` nor `createSelectors`. It would help someone to give feedback. A reproduction with as few lines as possible would be nice.

**Comment by dbritto-dev** [comment]
@wowblvck you need to be careful with namespace slices. btw, would you mind sharing us a minimal repro on stackblitz?

**Comment by wowblvck** [comment]
@dai-shi @dbritto-dev 

I reproduced an approximate scenario for how the store works in my application ([stackblits reproduce](https://stackblitz.com/edit/zustand-persist-error?file=src%2FApp.tsx))

The first time you run it, everything will work as expected, but when the page is reloaded, the slice methods are lost. To get them to work you need to clear your local storage.

@dai-shi Yes, this problem only occurs when I enable `persist`. Otherwise everything works correctly (at least I didn't notice any problems).

**Comment by wowblvck** [comment]
Okay, I realized that I need to use deepMerge in order to implement the storage methods.

Can you show how persist can be typed and also how methods can be injected if I use partialize in persist
